src/udev/scsi_id/scsi_id.c

```diff
@@ -136,12 +137,13 @@ static int get_file_options(const char *vendor, const char *model,
 
                 vendor_in = model_in = options_in = NULL;
 
-                buf = fgets(buffer, MAX_BUFFER_LEN, f);
-                if (!buf)
+                r = read_line(f, MAX_BUFFER_LEN, &buffer);
```

Comment:
Hmm? `read_line()` allocates the memory, so the buffer allocated in the above is leaked...

Reply:
agh of course, I don't know what I was thinking - fixed